rllib/algorithms/algorithm.py

```diff
@@ -2313,46 +2282,35 @@ def resolve_tf_settings():
 
     @OverrideToImplementCustomLogic_CallToSuperRecommended
     @DeveloperAPI
-    def validate_config(self, config: AlgorithmConfigDict) -> None:
-        """Validates a given config dict for this Algorithm.
+    def validate_config(
+        self,
+        config: Union[AlgorithmConfig, AlgorithmConfigDict],
+    ) -> None:
+        """Validates a given config object (or dictionary) for this Algorithm.
 
         Users should override this method to implement custom validation
         behavior. It is recommended to call `super().validate_config()` in
         this override.
 
         Args:
-            config: The given config dict to check.
+            config: The given config object (or dictionary) to check.
 
         Raises:
             ValueError: If there is something wrong with the config.
         """
-        model_config = config.get("model")
-        if model_config is None:
-            config["model"] = model_config = {}
-
-        # Use DefaultCallbacks class, if callbacks is None.
-        if config["callbacks"] is None:
-            config["callbacks"] = DefaultCallbacks
-        # Check, whether given `callbacks` is a callable.
-        if not callable(config["callbacks"]):
-            raise ValueError(
-                "`callbacks` must be a callable method that "
-                "returns a subclass of DefaultCallbacks, got "
-                f"{config['callbacks']}!"
-            )
+        from ray.rllib.models.catalog import MODEL_DEFAULTS
```

Comment:
A lot of the code here was moved into AlgorithmConfig itself. We should eventually deprecate all `validate_config` methods.